clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1072,6 +1067,52 @@ public int numStagedReceives(KafkaChannel channel) {
         return deque == null ? 0 : deque.size();
     }
 
+    class SelectorChannelMetadataRegistry implements ChannelMetadataRegistry {
+        private CipherInformation cipherInformation;
+        private ClientInformation clientInformation;
+
+        @Override
+        public void registerCipherInformation(final CipherInformation cipherInformation) {
```

Comment:
It would be good to make this consistent with `registerClientInformation` by making it re-register the cipher information if appropriate (and if the new information is different).
Question for @rajinisivaram: can the SSL cipher of a connection change when an SSL handshake is 
done on an already established SSL connection?

Reply:
I don't think the cipher could change but I will do it to stay consistent.